modules/api/pkg/resources/machine/common.go

```diff
@@ -619,6 +618,16 @@ func GetKubevirtProviderConfig(cluster *kubermaticv1.Cluster, nodeSpec apiv1.Nod
 		},
 	}
 
+	if dc.Spec.Kubevirt.EnableDedicatedCPUs {
+		vcpus, err := strconv.ParseInt(nodeSpec.Cloud.Kubevirt.CPUs, 0, 64)
+		if err != nil {
+			return nil, err
+		}
+		config.VirtualMachine.Template.VCPUs.Cores = int(vcpus)
+	} else {
+		config.VirtualMachine.Template.CPUs = providerconfig.ConfigVarString{Value: nodeSpec.Cloud.Kubevirt.CPUs}
+	}
+
```

Comment:
If a flag on dc should enforce such behaviour then it needs to be used on the UI as well with enough description(tooltips etc) that `EnableDedicatedCPUs` is enabled and `VCPUs` will be configured instead of `CPUs`.

Reply:
Why is that ? We have a lots of fields in the seed which are not using any tooltip and they have the exact same effect, actually we even have a whole mode that changes the whole behavior of the cloud provider(NamespacedMode) which is also not having tool tip or something similar. So I think this is also ok in this case

Reply:
NamespacedMode is used to determine the infra namespace and has nothing to do with the UI. This however is a field that is present in the machine deployment.

Reply:
I am not sure how this is any different tbh. You have a field on the seed that effects the handling of cloud provider resources and where they should be located(VMs, Policies, storage classes, PVCs) and you have another field on the seed that changes handling of the VMs(shared or dedicated CPUs). Both handling is happening on the cloud provider level. So how this is any different? And also, NamespcaedMode effects VPCs and Subnets, which are also UI components, and yet no tooltip for that either! So why now we should start using these tooltips 😉